### DISCUSSION TITLE
Unable to list elements

### DISCUSSION THREAD

**Discussion by breezko** [pocketbase/pocketbase#3228] [Q&A] [answered]
I have the following dataset, where `userId` is the id of the authenticated User from `users.id`

![image](https://github.com/pocketbase/pocketbase/assets/10911394/00f425d4-94d8-408d-875b-1111902a6e73)

This is my List/Search filter:
![image](https://github.com/pocketbase/pocketbase/assets/10911394/ac43b12b-e206-45b2-9a94-80fe3a981984)

However, when I try fetch the elements, that match my User ID I get 0 results.
```
final records = await pocketBase.collection('characters'). getFullList(
            sort: '-created',
          );
print("Get character records with auth ID: ${pocketBase.authStore.model.id}: $records");
```

`>> Get character records with auth ID: iu7xa5ky0ubarpc: []`

My expected result is to get the Character Element with the ID `kfub0sssa89gdjc`.

Additionally, when the `name` field is "unique", as in only one `users.id` has this name set then the result is correctly 1.

I am not sure if this is the correct place (whether it's an Pocketbase Rule issue, or Dart SDK related).

**Comment by ganigeorgiev** [comment, marked as answer]
The `@collection.*` api rule should be used only if you want to join and target external and non-related collection fields (_you can find some explanation about the rules syntax in https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax_).

If I'm understanding correctly your description, this is not the case here. You can replace the rule with:
```js
@request.auth.id = userId
```

**Reply by ganigeorgiev** [reply]
_Side-note: based on the screenshot it seems that the `userId` field is a `text` field and not a `relation` field. You can "convert" it (delete the existing and create a new one) to a `relation` field if you want to have user id validations and cascade operations._

**Reply by breezko** [reply]
Thanks! This actually works.

Yeah I previously had this relation field targeting to my users collection. So I was playing around with stupidly "copying" the users.id over to the characters collection.

`@request.auth.id = field.id` works now as intended.

**Reply by ganigeorgiev** [reply]
_Note that for single relations you can use directly just `field` instead of `field.id` (this is a minor optimization as it will avoid one unnecessary join operation; in the future this will be handled automatically behind the scenes by the server and will not matter)_

**Reply by breezko** [reply]
Awesome! Is there any more in-depth documentation than here https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax
or did I just didn't read correctly?

**Reply by ganigeorgiev** [reply]
No, there is no other documentation at the moment.

**Reply by breezko** [reply]
Ah... I would help on that, but this would require a deeper understanding, which I currently don't have, sadly.

**Comment by lavvy** [comment]
Hello please i just want make sure i am not crazy. I have installed the latest pocketbase ver. 0.22.16 and dart sdk ver 0.18.1.
But i cannot be able to filter out records belonging to each user in a collection. 



`final records = await pb
      .collection('configuration')
      .getFullList(filter: "userid = ${pb.authStore.model.id}");

  print("Get records with this auth ID: ${pb.authStore.model.id}: $records");`


It always pulls all the records without filtering anything. i have tried all solutions i saw on the internet, i just want to know that i'm not crazy

**Reply by ganigeorgiev** [reply]
I don't think your issue is related to the original question.

Based on the provided code sample it seems that you are missing quotes around the interpolated variable, aka. it needs to be:

```dart
filter: "userid = '${pb.authStore.model.id}'"

// or if you don't want to deal with manual quotes,
// check https://github.com/pocketbase/dart-sdk#binding-filter-parameters
filter: pb.filter("userid = {:id}", { "id": pb.authStore.model.id }
```

If you need further assistance please open a new Q&A post with more details about your use case, export of your collections configuration (_Admin UI > Settings > Export collections_) and a more detailed code sample how you are sending the request to reproduce the issue.
